cirq/experiments/cross_entropy_benchmarking.py

```diff
@@ -14,14 +14,41 @@
 
 from typing import (Any, Dict, Iterable, List, NamedTuple, Optional, Sequence,
                     Set, Tuple, Union)
+import dataclasses
 import numpy as np
+import scipy
 from matplotlib import pyplot as plt
 from cirq import circuits, devices, ops, protocols, sim, work
 
 CrossEntropyPair = NamedTuple('CrossEntropyPair', [('num_cycle', int),
                                                    ('xeb_fidelity', float)])
 
 
+@dataclasses.dataclass
+class DepolarizingModel:
+    """A model of depolarizing noise.
+
+    A model of a depolarizing process that maps a density matrix ρ as
+
+        ρ → (S * p^d) ρ + (1 - S * p^d) I / D
```

Comment:
The comment and formula mix two independent ideas. First, we use the depolarizing channel to model errors

    \rho -> p_eff \rho + (1 - p_eff) I / D

where p_eff is the effective decay constant of the channel. Then, we separate p_eff into SPAM errors which are depth-independent and circuit execution errors (decoherence and control errors) which exhibit exponential decay with depth. The separation works by fitting p_eff obtained for multiple circuit depths to

    p_eff = S * p^d

Note that we exploit the fact that the composition of depolarizing channel with decay constant a with depolarizing channel with decay constant b yields depolarizing channel with decay constant a*b.

Suggestion: perhaps describe the two ideas (how to model errors and how to separate SPAM and execution errors) separately.

Reply:
I've updated the documentation with your suggestions and correspondingly renamed the class to be specific to cross-entropy benchmarking.